### PR TITLE
Assign users to cards

### PR DESCRIPTION
**Pull request by juliushaertl**
bildschirmfoto vom 2017-10-01 17-13-40

This implements #11 

TODO:
- [x] Show avatars in board view
- [x] Show only users that are not already assigned
- [x] ~Notify only assigned users (requires #309)~ Lets notify all for now
- [x] Fix tests

### CODE REVIEW

**Comment by juliushaertl** [comment]
This is ready for testing/review. @pixelipo @artemanufrij Would be cool if you could give it a try.

Also cc @terrar @ossie @poVoq

File: templates/part.board.sidebarView.php [hunk 1]
```diff
@@ -18,7 +18,9 @@
 <div class="tabsContainer">
     <div id="commentsTabView" class="tab commentsTabView" ng-if="status.boardtab==0 || !status.boardtab">
 
-        <ui-select ng-if="boardservice.canShare()" ng-model="status.addSharee" theme="select2" style="width:100%;" title="Choose a user to assign" placeholder="Assign users ..." on-select="aclAdd(status.addSharee)" search-enabled="true">
+        <ui-select ng-if="boardservice.canShare()" ng-model="status.addSharee" theme="select2" style="width:100%;"
+				   title="Choose a user to assign" placeholder="Assign users ..."
```
Comment (pixelipo) [inline templates/part.board.sidebarView.php]: missing translation strings

File: templates/part.board.sidebarView.php [hunk 2]
```diff
@@ -18,7 +18,9 @@
 <div class="tabsContainer">
     <div id="commentsTabView" class="tab commentsTabView" ng-if="status.boardtab==0 || !status.boardtab">
 
-        <ui-select ng-if="boardservice.canShare()" ng-model="status.addSharee" theme="select2" style="width:100%;" title="Choose a user to assign" placeholder="Assign users ..." on-select="aclAdd(status.addSharee)" search-enabled="true">
+        <ui-select ng-if="boardservice.canShare()" ng-model="status.addSharee" theme="select2" style="width:100%;"
```
Comment (pixelipo) [inline templates/part.board.sidebarView.php]: is the inline style (`width`) really needed here?

File: css/style.scss [hunk 3]
```diff
@@ -598,29 +604,32 @@ input.input-inline {
...
-		font-size: 15px;
-
-		> div {
-			display: inline-flex;
-			align-content: center;
-			justify-content: space-between;
+		display: flex;
+		margin-bottom: 5px;
+		margin-top: 20px;
+		min-height: 30px;
```
Comment (pixelipo) [inline css/style.scss]: is `min-height` really needed?

File: css/style.scss [hunk 4]
```diff
@@ -598,29 +604,32 @@ input.input-inline {
...
+		h4 {
+			padding-top: 5px;
+			padding-bottom: 5px;
+			border: none;
```
Comment (pixelipo) [inline css/style.scss]: not needed?

File: css/style.scss [hunk 5]
```diff
@@ -598,29 +604,32 @@ input.input-inline {
...
+			flex: 1;
 		}
+
```
Comment (pixelipo) [inline css/style.scss]: Empty lines

**Comment by pixelipo** [comment]
@juliushaertl I made a couple of commits here, I hope you don't mind. I think they make the whole `.card-controls` bar a bit cleaner and more consistent:

- it's actually 2px smaller now, to a standard 44px used throughout NC
- `.icon-more` icon is now more touchscreen-friendly, with and activation area of 44x44px
- avatars are 32px standard size without affecting overall height

**Comment by juliushaertl** [comment]
Thanks for the commits. There is a small issue when no card description is set. I'll take care of that.

**Comment by juliushaertl** [comment]
@pixelipo Fixed. Thanks you.

**Comment by ossie** [comment]
@juliushaertl I checked out pr #314 and tried to assign users: Fetching the user list fails. 
I keep getting JS errors: `[$injector:unpr] http://errors.angularjs.org/1.6.4/$injector/unpr?p0=withoutAssignedUsersFilterProvider%20%3C-%20withoutAssignedUsersFilter`. 
Users assigned directly in the database look fine, though.

**Comment by juliushaertl** [comment]
@ossie Thanks for giving it a try. That migh be caused when trying it before https://github.com/nextcloud/deck/pull/314/commits/9b54d1c9ef855d7ffaf5d295e471ded9ac5d7bb5#diff-70c43e8d89ea7f7d5e64f83246f17ed0R50 was added to the PR, and running in debug mode. At least i could not reproduce it anymore. Let's merge now. Feel free to open an issue if that sill occurs on master.

**Pull request merged**